binaries/cuprated/src/rpc/handler.rs

```diff
@@ -3,51 +3,136 @@
 use std::task::{Context, Poll};
 
 use anyhow::Error;
-use futures::{channel::oneshot::channel, future::BoxFuture};
-use serde::{Deserialize, Serialize};
+use futures::future::BoxFuture;
+use monero_serai::block::Block;
 use tower::Service;
 
-use cuprate_blockchain::service::BlockchainReadHandle;
-use cuprate_helper::asynch::InfallibleOneshotReceiver;
-use cuprate_json_rpc::Id;
+use cuprate_blockchain::service::{BlockchainReadHandle, BlockchainWriteHandle};
 use cuprate_rpc_interface::RpcHandler;
 use cuprate_rpc_types::{
     bin::{BinRequest, BinResponse},
     json::{JsonRpcRequest, JsonRpcResponse},
     other::{OtherRequest, OtherResponse},
 };
-use cuprate_txpool::service::TxpoolReadHandle;
+use cuprate_txpool::service::{TxpoolReadHandle, TxpoolWriteHandle};
 
 use crate::rpc::{bin, json, other};
 
-/// TODO
+/// TODO: use real type when public.
 #[derive(Clone)]
-pub struct CupratedRpcHandler {
-    /// Should this RPC server be [restricted](RpcHandler::restricted)?
-    //
-    // INVARIANT:
-    // We don't need to include this in `state` and check for
-    // `self.is_restricted()` because `cuprate-rpc-interface` handles that.
-    pub restricted: bool,
-
-    /// State needed for request -> response mapping.
-    pub state: CupratedRpcHandlerState,
+#[expect(clippy::large_enum_variant)]
+pub enum BlockchainManagerRequest {
+    /// Pop blocks off the top of the blockchain.
+    ///
+    /// Input is the amount of blocks to pop.
+    PopBlocks { amount: usize },
+
+    /// Start pruning the blockchain.
+    Prune,
+
+    /// Is the blockchain pruned?
+    Pruned,
+
+    /// Relay a block to the network.
+    RelayBlock(Block),
```

Comment:
This should be add block, this also happens to be the one request I already have in the blockchain manager PR (`handle_incoming_block`).

Reply:
Does it have the same semantics though? It adds a block to our local chain _and_ broadcasts it?

The usecase is [`submit_block`](https://github.com/monero-project/monero/blob/9866a0e9021e2422d8055731a586083eb5e2be67/src/rpc/core_rpc_server.cpp#L2218).

Reply:
yep